src/main/java/org/jabref/gui/preferences/AppearanceTabViewModel.java

```diff
@@ -53,55 +63,95 @@ public AppearanceTabViewModel(DialogService dialogService, JabRefPreferences pre
                         Localization.lang("Appearance"),
                         Localization.lang("Font settings"),
                         Localization.lang("You must enter an integer value higher than 8."))));
+
+        customPathToThemeValidator = new FunctionBasedValidator<>(
+                customPathToThemeProperty,
+                input -> !StringUtil.isNullOrEmpty(input),
+                ValidationMessage.error(String.format("%s > %s %n %n %s",
+                        Localization.lang("Appearance"),
+                        Localization.lang("Visual theme"),
+                        Localization.lang("Please specify a css theme file."))));
     }
 
     @Override
     public void setValues() {
-        fontOverrideProperty.setValue(preferences.getBoolean(JabRefPreferences.OVERRIDE_DEFAULT_FONT_SIZE));
-        fontSizeProperty.setValue(String.valueOf(preferences.getInt(JabRefPreferences.MAIN_FONT_SIZE)));
-
-        switch (preferences.get(JabRefPreferences.FX_THEME)) {
-            case ThemeLoader.DARK_CSS:
-                themeLightProperty.setValue(false);
-                themeDarkProperty.setValue(true);
-                break;
-            case ThemeLoader.MAIN_CSS:
-            default:
-                themeLightProperty.setValue(true);
-                themeDarkProperty.setValue(false);
+        fontOverrideProperty.setValue(initialAppearancePreferences.shouldOverrideDefaultFontSize());
+        fontSizeProperty.setValue(String.valueOf(initialAppearancePreferences.getMainFontSize()));
+
+        String currentTheme = initialAppearancePreferences.getPathToTheme();
```

Comment:
I'm not sure if this makes things clearer, but just wanted to throw the idea in: What about having an enum for the different kinds of themes (light / dark / custom), and then an additional `Path getPathToCustomTheme` in case the "custom" enum option is chosen. Right now it feels like `getPathToTheme` is mixing these two aspects.

Reply:
Yeah, possible, but It escalated a bit. Again. :see_no_evil: